src/olympia/promoted/tasks.py

```diff
@@ -106,3 +110,66 @@ def on_stripe_charge_failed(event):
         from_email=settings.ADDONS_EMAIL,
         recipient_list=[settings.VERIFIED_ADDONS_EMAIL],
     )
+
+
+@task
+@use_primary_db
+def on_stripe_customer_subscription_deleted(event):
+    event_id = event.get("id")
+    event_type = event.get("type")
+
+    if event_type != "customer.subscription.deleted":
+        log.error(
+            'invalid event "%s" received (event_id=%s).', event_type, event_id
+        )
+        return
+
+    # This event should contain a `subscription` object.
+    subscription = event.get("data", {}).get("object")
+    if not subscription:
+        log.error("no subscription object in event (event_id=%s).", event_id)
+        return
+
+    subscription_id = subscription["id"]
+    log.info(
+        'received "%s" event with subscription_id=%s.',
+        event_type,
+        subscription_id,
+    )
+
+    try:
+        sub = PromotedSubscription.objects.get(
+            stripe_subscription_id=subscription_id
+        )
+    except PromotedSubscription.DoesNotExist:
+        log.exception(
+            'received a "%s" event (event_id=%s) for a non-existent'
+            " promoted subscription (subscription_id=%s).",
+            event_type,
+            event_id,
+            subscription_id,
+        )
+        return
+
+    if sub.cancelled_at:
+        log.info(
+            "PromotedSubscription %s already cancelled, ignoring event"
+            " (event_id=%s).",
+            sub.id,
+            event_id,
+        )
+        return
+
+    sub.update(
+        cancelled_at=datetime.datetime.fromtimestamp(
+            subscription["canceled_at"]
+        )
+    )
+    log.info("PromotedSubscription %s has been marked as cancelled.", sub.id)
+
+    # TODO: this should trigger resigning somehow, see:
+    # https://github.com/mozilla/addons-server/issues/15921
+    sub.promoted_addon.update(group_id=NOT_PROMOTED.id)
+    log.info(
+        "PromotedAddon %s is not promoted anymore.", sub.promoted_addon.id
```

Comment:
I'd lean towards saying we should be logging `Addon.id` here, rather then `PromotedAddon.id`, but is that consistent with other logging you've added?

Reply:
I added the addon id, the promoted addon id and the promoted sub id